src/content/docs/zh-cn/guides/cms/builderio.mdx

```diff
@@ -0,0 +1,361 @@
+    ></builder-component>
+    <script async src="https://cdn.builder.io/js/webcomponents"></script>
+
+    <footer>这里是页脚</footer>
+  </body>
+</html>
+
+```
+
+在上面的例子中，`<builder-component>` 元素指示 Builder 在页面的哪个位置插入 CMS 内容。
+
+#### 设置新路由作为预览 URL
+
+1. 复制完整的预览 URL，要包括协议，复制到你的剪贴板（例如 `https://{your host}/builder-preview`）。
+
+2. 进入你的 Builder 空间的 **模型** 标签页，选择你创建的模型，并将步骤 1 中的 URL 粘贴到 **预览 URL** 字段。确保 URL 是完整的，复制要包括协议，例如 `https://`。
+
+3. 点击右上角的 **保存（Save）** 按钮。
+
+:::tip
+当你的网站部署到生产环境时，记得将预览 URL 更新为与你的正式网站地址相匹配，例如 `https://myAwesomeAstroBlog.com/builder-preview`。
+:::
+
+#### 测试预览 URL 设置
+
+1. 确保你的开发服务器正在运行，且网站在线。同时，确认 `/builder-preview` 路由能够正常工作。
+
+2. 在你的 Builder 空间中，找到 **内容（Content）** 标签页，点击 **新建（New）** 来创建一个新的 `blogpost` 模型的内容条目。
+
+3. 在新打开的 Builder 编辑器中，你应该能看到 `builder-preview.astro` 页面，页面中间有一个显眼的 **添加块（Add Block）**。
+
+:::tip[故障排除]
+
+在设置预览时，可能会遇到一些问题。如果出现任何异常，你可以尝试以下步骤：
+
+* 确保网站处于在线状态 - 比如，你的开发服务器正在运行。
+* 确保 URL 完全一致 - 你的 Astro 项目中的 URL 应与在 Builder 应用中设置的 URL 完全匹配。
+* 确保使用的是完整的 URL，包括协议，如 `https://`。
+* 如果你在像 [StackBlitz](https://stackblitz.com/) 或 [Gitpod](https://www.gitpod.io/) 这样的虚拟环境中工作，每次重启工作区时可能需要重新复制并粘贴 URL，因为这些环境通常会为你的项目生成一个新的 URL。
+
+了解更多信息，请参阅 [Builder 的故障排除指南](https://www.builder.io/c/docs/guides/preview-url-working)。
+:::
+
+### 创建博客文章
+
+1. 在 Builder 的可视化编辑器中，创建一个新的内容条目，值如下：
+    - **标题：** '第一篇文章，哇哦！'
+    - **slug：** 'first-post-woohoo'
+2. 利用 **添加块（Add Block）** 功能来丰富你的文章内容，添加所需的文本字段。
+3. 在编辑器顶部的文本框中为你的内容条目命名，这个名称将用作在 Builder 应用中的显示名称。
+4. 准备就绪后，点击右上角的 **发布（Publish）** 按钮。
+5. 你可以随意创建更多文章，但需要确保每篇文章都有一个 `title` 和一个 `slug` 字段，以及相应的文章内容。
+
+### 显示博客文章列表
+
+在 `src/pages/index.astro` 文件中添加以下代码，以便展示一个包含所有文章标题的列表，其中每个标题都链接到相应的文章页面：
+
+```astro title="src/pages/index.astro" {9}
+---
+
+const builderAPIpublicKey = import.meta.env.BUILDER_API_PUBLIC_KEY;
+const builderModel = import.meta.env.BUILDER_BLOGPOST_MODEL;
+
+const { results: posts } = await fetch(
+  `https://cdn.builder.io/api/v3/content/${builderModel}?${new URLSearchParams({
+    apiKey: builderAPIpublicKey,
+    fields: ["data.slug", "data.title"].join(","),
+    cachebust: "true",
+  }).toString()}`
+)
+  .then((res) => res.json())
+  .catch();
+---
+
+<html lang="en">
+  <head>
+    <title>博客目录</title>
+  </head>
+  <body>
+    <ul>
+      {
+        posts.map(({ data: { slug, title } }) => (
+          <li>
+            <a href={`/posts/${slug}`}>{title}</a>
+          </li>
+        ))
+      }
+    </ul>
+  </body>
+</html>
+
+```
+
+我们通过内容 API 获取了一个包含每篇文章数据的对象数组。`fields` 查询参数用于指定我们希望 Builder 返回哪些数据字段（如高亮代码所示）。`slug` 和 `title` 应与你在 Builder 模型中添加的自定义数据字段名称相匹配。
+
+我们已经在首页上使用从 API 获取的 `posts` 数组显示了博客文章标题的列表。接下来，我们将创建用于显示单篇文章的页面路由。
+
+:::tip[框架集成]
+如果你的 Astro 项目中集成了 JavaScript 框架（如 Svelte，Vue 或 React），那么你可以考虑使用 [Builder 的集成功能](https://www.builder.io/m/integrations)，这是一种比直接通过 REST API 更优雅得获取数据的方法。
+:::
+
+访问你的索引路由，你将看到一个博客文章标题的链接列表！
+
+
+### 显示单个博客文章
+
+创建页面 `src/pages/posts/[slug].astro`，该页面将为每篇文章[动态生成一个页面](/zh-cn/core-concepts/routing/#动态路由)。
```

Comment:
On `main` we've updated this route to `/guides/`. You'll also need to update branch when you update this
```suggestion
创建页面 `src/pages/posts/[slug].astro`，该页面将为每篇文章[动态生成一个页面](/zh-cn/guides/routing/#动态路由)。
```